File: src/main/java/org/khun/leetcode/ArrayAndString/LongestCommonPrefix.java
package org.khun.leetcode.ArrayAndString;

import java.util.Arrays;

public class LongestCommonPrefix {
    public String longestCommonPrefix(String[] strs){
        Arrays.sort(strs);
        String first = strs[0];
        String last = strs[strs.length - 1];
        int i = 0;
        while (i<first.length() && i<last.length() && first.charAt(i) == last.charAt(i)) i++;
        return first.substring(0, i);
    }

    public static void main(String[] args) {
        LongestCommonPrefix longestCommonPrefix = new LongestCommonPrefix();

        String[] strs1 = {"flower", "flow", "flight"};
        System.out.println(longestCommonPrefix.longestCommonPrefix(strs1)); // Output: "fl"

        String[] strs2 = {"dog", "racecar", "car"};
        System.out.println(longestCommonPrefix.longestCommonPrefix(strs2)); // Output: ""
    }
}

File: src/main/java/org/khun/leetcode/ArrayAndString/LongestCommonPrefix.kt
package org.khun.leetcode.ArrayAndString

fun longestCommonPrefix(strs: Array<String>): String {
    strs.sort()
    var i = 0
    while (i<strs.first().length && i<strs.last().length && strs.first()[i] == strs.last()[i]) i++
    return strs.first().substring(0, i)
}

fun main() {
    val strs1 = arrayOf("flower", "flow", "flight")
    println(longestCommonPrefix(strs1)) // Output: "fl"

    val strs2 = arrayOf("dog", "racecar", "car")
    println(longestCommonPrefix(strs2))
}